### ISSUE TITLE
Community page does not load if repo doesn't have an auspice directory

### ISSUE THREAD

**Issue by joverlee521** [nextstrain/nextstrain.org#617]
### Current Behavior
If a GitHub repo has a narratives directory but does not include an auspice directory then the community page does not load. 

### Expected behavior
A community page should load even if the repository only contains narratives. 

### How to reproduce
Steps to reproduce the current behavior:
1. Go to https://nextstrain.org/community/Lschulz2022/nextstrain-narratives-demo
2. See error on page
3. Error in console shows https://next.nextstrain.org/charon/getAvailable?prefix=/community/Lschulz2022/nextstrain-narratives-demo/ returns 404.

**Comment by tsibley** [comment]
The console contains this for me:

![image](https://user-images.githubusercontent.com/79913/198370258-f0bce43f-dff4-46c7-9217-6365da2b5238.png)

which comes from the underlying error we're catching here:

https://github.com/nextstrain/nextstrain.org/blob/b68e829f52440a74862b0de1fb5eb10b4d07a208/static-site/src/sections/community-repo-page.jsx#L49-L50

is `TypeError: e is undefined`. In the debugger, it traces back to here:

https://github.com/nextstrain/nextstrain.org/blob/b68e829f52440a74862b0de1fb5eb10b4d07a208/static-site/src/sections/community-repo-page.jsx#L23-L25

where `list` (`e` after minification) is `undefined`, which I think we can fix with:

```diff
diff --git a/static-site/src/sections/community-repo-page.jsx b/static-site/src/sections/community-repo-page.jsx
index a81e1cc..f6e6321 100644
--- a/static-site/src/sections/community-repo-page.jsx
+++ b/static-site/src/sections/community-repo-page.jsx
@@ -21,6 +21,8 @@ class Index extends React.Component {
 
   // parse getAvailable listing into one that dataset-select component accepts
   createDatasetListing = (list, userName) => {
+    if (!list) return [];
+
     // Note that the `request` always includes @branch, irregardless of URL.
     return list.map((d) => {
       return {
```

**Comment by tsibley** [comment]
Oh, :facepalm:. That's not quite right. There's a server-side issue here too.

**Comment by tsibley** [comment]
The 404 is coming from here.

https://github.com/nextstrain/nextstrain.org/blob/b68e829f52440a74862b0de1fb5eb10b4d07a208/src/sources/community.js#L60-L64

 The CommunitySource class accounts for narratives not existing in the repo, but with that line requires datasets to exist. We can fix that.

**Comment by joverlee521** [comment]
> I think there's one open question about what we want the community repo page to do when the repo doesn't actually exist (vs. it exists but there's neither any datasets or narratives in it)

It might be helpful to display a link for the GitHub repo on the page so users can click on it to verify they are trying to access the correct repo? Instead of making nextstrain.org a proxy for GitHub, just let users check directly at the source if the repo exists or if data exists.

**Comment by jameshadfield** [comment]
> but I think there's one open question about what we want the community repo page to do when the repo doesn't actually exist

> it exists but there's neither any datasets or narratives in it

In both cases, I think turning the current error banner into a simple page explaining that the github repo X/Y either (a) doesn't exist or (b) exists but doesn't have any datasets or narratives in it would be ideal. We can then link to [our docs for community builds](https://docs.nextstrain.org/en/latest/guides/share/community-builds.html).

**Comment by tsibley** [comment]
Vexed by weird new errors for a bit until I tracked them down to the fact that [an "auspice" _file_](https://github.com/LSchulz2022/nextstrain-narratives-demo/blob/551a69a7becf17bd1b6f8cb6976a50fc44da1272/auspice) (not directory) has since been added to the example repo in the issue description. Presumably as a workaround for this bug!

**Comment by tsibley** [comment]
Should be resolved by #622.

**Comment by tsibley** [comment]
Fix deployed to production just a few minutes ago.
